### DISCUSSION TITLE
[AnalyticalTable] Tree table with hierachical data does not retain style on expand due to app level box-sizing: border-box

### DISCUSSION THREAD

**Discussion by kmanisha29** [UI5/webcomponents-react#6366] [Q&A] [answered]
I am using analytical table with tree view having hirarchiacal data.
On version 1.26, I see that 1st column with anchor tag for expand / collapse with custom cell or even normal cell is  not maintaining style for hierarchical; view. No left space being added.  

Not sure if exact cause but on investigating i found host application has used  *{box-sizing: border-box} at root level and when toggling from devtools for this style, nested space shows up ..
Could you please suggest how to handle this we can't remove box-sizing : border-box style from host app.
We are devloping micro frontend and only have control over that unit.
<img width="411" alt="Screenshot 2024-09-16 at 1 01 51 PM" src="https://github.com/user-attachments/assets/01f1cf9b-fcac-4650-8482-4c3b22c0e772">

Another issue: since v1.26, an extra space is getting added to all columns in tree table and it is an element due to which text is not getting aligned from beigining ( TextAlign.Begin). It looks like centrally aligned even horizontally.
<img width="1191" alt="Screenshot 2024-09-16 at 12 57 55 PM" src="https://github.com/user-attachments/assets/23a9b140-6861-4e53-8f6c-9259d6b0e5b8">
<img width="1728" alt="Screenshot 2024-09-16 at 12 58 06 PM" src="https://github.com/user-attachments/assets/78fa61b5-5adc-4bbf-b729-d5f9df5628da">

<img width="497" alt="Screenshot 2024-09-13 at 2 38 53 PM" src="https://github.com/user-attachments/assets/3642d571-e895-4e14-be6d-616e9acf2a79">

Attached stackbliz link with both issue shown: https://stackblitz.com/edit/github-3wiuce-kjl4sq?file=src%2FApp.tsx,src%2FTreeCell.tsx,src%2FuseData.ts

**Comment by Lukas742** [comment]
Hi @kmanisha29 

supporting overriding the default `box-sizing` for all elements is unfortunately not supported by our library, so I can't really offer a solution here.

For the indentation issue, I've added an answer here: https://github.com/SAP/ui5-webcomponents-react/issues/6368

**Comment by kmanisha29** [comment]
The style in host app is actually : 
html {
  box-sizing: border-box;
}
*,
*:before,
*:after {
  box-sizing: inherit;
}

So any other element that needs content-box can add it and inherit for its child element.
Could you please suggest if it should  be handled from the component side or a workaround can be provided in library.
We are using MFE to show this component in host app whose style we cannot change from root level.

**Reply by Lukas742** [reply]
We're using a [CSS layer](https://developer.mozilla.org/en-US/docs/Web/CSS/@layer) for all our styles, so if you're wrapping your global styles in another layer, the specificity is lower and therefore the styles of `@ui5/webcomponents-react` have precedence. 

Example: https://stackblitz.com/edit/ui5wcr-at-7strnz?file=src%2Findex.css,src%2FApp.tsx

Also, it's possible to explicitly import the CSS bundles by setting the `staticCSSInjected` prop of the `ThemeProvider` to `true`. ([docs](https://sap.github.io/ui5-webcomponents-react/v2/?path=/docs/knowledge-base-styling--docs#explicitly-import-css-bundles)). This way, you're in control when and how the CSS bundle/s are imported.

___

Currently, there's no way to configure whether a CSS layer is applied or not. We'll look into this, but it might take some time before we can address this topic.

**Reply by Lukas742** [reply, marked as answer]
Hi @kmanisha29 
we've discussed this topic internally and came to the conclusion that using a layer for our internal CSS should be the default. 
This does __not__ apply for the static bundles. In [v2.2.0](https://github.com/SAP/ui5-webcomponents-react/releases/tag/v2.2.0) we removed the layer from the generated classes again, so apps have now more control over the bundles. You can find out more about this [here](https://sap.github.io/ui5-webcomponents-react/v2/?path=/docs/knowledge-base-styling--docs#explicitly-import-css-bundles).
